docs/install_csi_driver.md

```diff
@@ -0,0 +1,174 @@
+# Nutanix CSI Driver installation with CAPX
+
+The Nutanix CSI driver is fully supported on CAPI/CAPX deployed clusters where nodes follow Nutanix prerequisites.
+
+You can install it yourself in a workload cluster by following the official documentation (Helm based) or let the management cluster install it for you using `ClusterResourceSet`.
+
+
+
+## CAPI Workload cluster prerequisites for Nutanix CSI Driver
+
+Kubernetes workers need the following prerequisite to use the Nutanix CSI Drivers:
+
+- iSCSI initiator package (for Volumes based block storage)
+- NFS client package ( for Files based storage)
+
+These packages can already be present in the image you use with your Infrastructure provider or you can also rely on your bootstrap provider to install them.
+
+The packages name and installation method will also vary depending on the OS you plan to use.
+
+In the example below, we will see how to use the `kubeadm` bootstrap provider to deploy these packages on top of an Ubuntu 20.04 image.
+
+The `kubeadm` bootstrap provider allows defining `preKubeadmCommands` that will be launched before kubernetes cluster creation.
+
+These `preKubeadmCommands` can be defined both in `KubeadmControlPlane` for master nodes and in `KubeadmConfigTemplate` for worker nodes.
+
+In our case with an Ubuntu 20.04 image, you need to modify both `KubeadmControlPlane` and `KubeadmConfigTemplate` as in the example below in order to install the prerequisites on all nodes.
+
+```yaml
+spec:
+  template:
+    spec:
+      # .......
+      preKubeadmCommands:
```

Comment:
can we make this part of the csi flavor of the cluster-template.yaml? say by creating cluster-template-nutanix-csi.yaml